onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -430,14 +464,18 @@ Status CUDAExecutionProvider::OnRunEnd(bool sync_stream, const onnxruntime::RunO
 }
 
 bool CUDAExecutionProvider::IsGraphCaptureEnabled() const {
-  return info_.enable_cuda_graph;
+  return info_.enable_cuda_graph && GetPerThreadContext().IsGraphCaptureAllowedOnRun();
```

Comment:
I think the EP level `IsGraphCaptureEnabled()` interface need not depend on whether on graph capture is allowed on the run or not? It should simply return if the EP is setup to do graph capture or not - whether individual Run() allows it or not can be handled `GetPerThreadContext().IsGraphCaptureAllowed()` (which is the current code).

Reply:
it's added by accident. removed